llama_index/evaluation/multi_modal/faithfulness.py

```diff
@@ -0,0 +1,222 @@
+            The template to use for evaluation.
+        refine_template(Optional[Union[str, BasePromptTemplate]]):
+            The template to use for refining the evaluation.
+    """
+
+    def __init__(
+        self,
+        multi_modal_llm: MultiModalLLM | None = None,
+        raise_error: bool = False,
+        eval_template: str | BasePromptTemplate | None = None,
+        refine_template: str | BasePromptTemplate | None = None,
+    ) -> None:
+        """Init params."""
+        self._multi_modal_llm = multi_modal_llm or OpenAIMultiModal(
+            model="gpt-4-vision-preview", max_new_tokens=1000
+        )
+        self._raise_error = raise_error
+
+        self._eval_template: BasePromptTemplate
+        if isinstance(eval_template, str):
+            self._eval_template = PromptTemplate(eval_template)
+        else:
+            self._eval_template = eval_template or DEFAULT_EVAL_TEMPLATE
+
+        self._refine_template: BasePromptTemplate
+        if isinstance(refine_template, str):
+            self._refine_template = PromptTemplate(refine_template)
+        else:
+            self._refine_template = refine_template or DEFAULT_REFINE_TEMPLATE
+
+    def _get_prompts(self) -> PromptDictType:
+        """Get prompts."""
+        return {
+            "eval_template": self._eval_template,
+            "refine_template": self._refine_template,
+        }
+
+    def _update_prompts(self, prompts: PromptDictType) -> None:
+        """Update prompts."""
+        if "eval_template" in prompts:
+            self._eval_template = prompts["eval_template"]
+        if "refine_template" in prompts:
+            self._refine_template = prompts["refine_template"]
+
+    def evaluate(
+        self,
+        query: str | None = None,
+        response: str | None = None,
+        contexts: Sequence[str] | None = None,
+        image_paths: List[str] | None = None,
+        image_urls: List[str] | None = None,
+        **kwargs: Any,
+    ) -> EvaluationResult:
+        """Evaluate whether the response is faithful to the multi-modal contexts."""
+        del query  # Unused
+        del kwargs  # Unused
+        if contexts is None or response is None:
+            raise ValueError("contexts and response must be provided")
+
+        context_str = "\n\n".join(contexts)
+        fmt_prompt = self._eval_template.format(
+            context_str=context_str, query_str=response
+        )
+
+        image_documents = []
+        if image_paths:
+            for image_path in image_paths:
+                image_documents += SimpleDirectoryReader(
+                    input_files=[image_path]
+                ).load_data()
+            node_parser = SentenceSplitter.from_defaults()
+            image_nodes = node_parser.get_nodes_from_documents(image_documents)
+        if image_urls:
+            ...
+
+        response_obj = self._multi_modal_llm.complete(
+            prompt=fmt_prompt,
+            image_documents=image_nodes,
```

Comment:
Hmm if you passed multiple `image_paths` this would pass every image to the LLM in one step and generate a single response?